src/platform/PICO/dma_pico.c

```diff
@@ -24,13 +24,106 @@
 #include <string.h>
 
 #include "platform.h"
+
+#ifdef USE_DMA
+
 #include "drivers/dma.h"
-#include "dma_pico.h"
 
-dmaChannelDescriptor_t dmaDescriptors[DMA_HANDLER_END-1] = {
-    
+#include "platform/dma.h"
+#include "pico/multicore.h"
+#include "hardware/dma.h"
+#include "hardware/irq.h"
+
+volatile bool dma_irq0_handler_registered = false;
+volatile bool dma_irq1_handler_registered = false;
+
+dmaChannelDescriptor_t dmaDescriptors[DMA_LAST_HANDLER] = {
```

Comment:
Isn't is possible to use `dmaDescriptors[]` It's a bit more resilient.
Also using `DMA_LAST_HANDLER` to compensate for 1-based indexing is confusing